manifests/plugin/openscap.pp

```diff
@@ -0,0 +1,63 @@
+# = Foreman OpenSCAP plugin
+#
+# This class installs OpenSCAP plugin
+#
+# === Parameters:
+#
+# $configure_openscap_repo::  Enable custom yum repo with packages needed for foreman_openscap,
+#                             type:boolean
+#
+# $scap_client_module_dir::   if you set a directory here, installer will upload foreman_scap_client
+#                             puppet module into this directory for you
+#                             e.g. /etc/puppet/environments/production/modules
+#
+class foreman::plugin::openscap (
+  $configure_openscap_repo = false,
+  $scap_client_module_dir  = undef,
+) {
+  validate_bool($configure_openscap_repo)
+
+  case $::osfamily {
+    'RedHat': {
+
+      if $configure_openscap_repo {
+        case $::operatingsystem {
+          'fedora': {
+            $repo = 'fedora'
+          }
+          default: {
+            $repo = 'epel'
+          }
+        }
+
+        if $::operatingsystemmajrelease == undef {
+          $versions_array = split($::operatingsystemrelease, '\.') # facter 1.6
+          $major = $versions_array[0]
+        } else {
+          $major = $::operatingsystemmajrelease # facter 1.7+
+        }
+
+        yumrepo { 'isimluk-openscap':
+          enabled  => 1,
+          gpgcheck => 0,
+          baseurl  => "http://copr-be.cloud.fedoraproject.org/results/isimluk/OpenSCAP/${repo}-${major}-\$basearch/",
+          before   => [ Foreman::Plugin['openscap'] ],
+        }
+      }
+
+      foreman::plugin {'openscap': }
+
+      if $scap_client_module_dir {
+        file { "${scap_client_module_dir}/foreman_scap_client":
+          ensure  => directory,
+          source  => '/usr/share/foreman-installer/modules/foreman_scap_client',
```

Comment:
In order to configure scap clients to run scans and upload it to foreman we have to setup foreman_scap_client script. foreman-openscap plugin rely on existing foreman_scap_client puppet class which it assigns to specified hostgroup in order to configure foreman_scap_client script. Basically foreman-openscap gathers information about policies and content from wizards and apply it on host through this puppet class which sets a cron and other things so that foreman_scap_client is run.  
We wanted to make it easy for users to use foreman-openscap so they don't have to install foreman_scap_client puppet module manually. Installer could do that for them if they explicitly ask for it and they select an environment where it should be put.  
Please ask if I should elaborate some part.